concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs

```diff
@@ -605,6 +634,65 @@ processBakerStakeThresholdUpdates t bu = do
                     pendingUpdates = pendingUpdates{pBakerStakeThresholdQueue = newpQ}
                 }
 
+-- |Process the add anonymity revoker update queue.
+--  Ignores updates with duplicate ARs.
+processAddAnonymityRevokerUpdates :: (MonadBlobStore m)
+  => Timestamp
+  -> BufferedRef Updates
+  -> HashedBufferedRef ARS.AnonymityRevokers
+  -> m (Map.Map TransactionTime UpdateValue, BufferedRef Updates, HashedBufferedRef ARS.AnonymityRevokers)
+processAddAnonymityRevokerUpdates t bu hbar = do
+    u@Updates{..} <- refLoad bu
+    oldARs <- ARS.arRevokers <$> refLoad hbar
+    oldQ <- refLoad (pAddAnonymityRevokerQueue pendingUpdates)
+    let (ql, qr) = Seq.spanl ((<= t) . transactionTimeToTimestamp . fst) (uqQueue oldQ)
+    (changes, updatedARs) <- addAndAccumNonduplicateUpdates oldARs ARS.arIdentity UVAddAnonymityRevoker ql
```

Comment:
Why does this not return the `AnonymityRevokers` structure directly (instead of the Map it returns now)?

Reply:
The reason is that it is used for both `AnonymityRevokers` and `IdentityProviders`.
Perhaps there is some nice way to unwrap and wrap these newtypes without adding more parameters to the function?